test/functional/apps/visualize/_tsvb_markdown.js

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import expect from 'expect.js';
+
+export default function ({ getService, getPageObjects }) {
+  const retry = getService('retry');
+  const find = getService('find');
+  const { visualBuilder, timePicker } = getPageObjects(['common', 'visualize', 'header', 'settings', 'visualBuilder', 'timePicker']);
+
+  describe('visual builder', function describeIndexTests() {
+
+    describe('markdown', () => {
+
+      before(async () => {
+        await visualBuilder.resetPage();
+        await visualBuilder.clickMarkdown();
+        await timePicker.setAbsoluteRange('2015-09-22 06:00:00.000', '2015-09-22 11:00:00.000');
+      });
+
+      it('should be render all markdown components', async () => {
+        const table = await find.byCssSelector('table.table');
+        const isExisting = find.exists(table);
+        const tabsCount = await visualBuilder.getSubTabsCount();
```

Comment:
1. The general rule for functional tests: use `testSubjects` over `find` service as much as possible.
In particular case, you need to add `data-test-subj="tsvbMarkdownVariablesTable"` for table here:
https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/metrics/public/components/markdown_editor.js#L172
Then you can use `testSubjects.find('tsvbMarkdownVariablesTable')`

2. Verifying table is present is **not good enough**: it can be empty as screenshot displays
<img width="1336" alt="tsvb_markdown" src="https://user-images.githubusercontent.com/10977896/52630756-633ab800-2ebd-11e9-81e6-23ae44b199d8.png">
This test should check variables list is rendered as it was frequent issue in the past

3. Avoid direct usage of `testSubject`/`find` services in test. Should be wrapped up in PageObject:
```
const variablesList = visualBuilder.getMarkdownTableVariables()
expect(variablesList).to.have.length(5)
```